### ISSUE TITLE
Feature Request: Background Rendering

### ISSUE THREAD

**Issue by Pyrif** [LMMS/lmms#4162]
It would be optimal to allow LMMS to render in the background and free up CPU for other programs, it can get rather annoying when LMMS is rendering something and I'm trying to type a message and my letters fall into the text box in chunks rather than one-by-one!

**Comment by lukas-w** [comment]
@Pyrif What OS are you using? Your system shouldn't become unresponsive due to a single process. You can try lowering LMMS's priority in the system's task/process manager and see if that helps.

**Comment by Pyrif** [comment]
I'm requesting it as a feature because it's a good idea to implement it as one. Not all users know how to use Task Manager and know how/are willing to change their tasks' priorities. It's a feature built into many other DAWs, and should likely be a part of LMMS as well.

**Comment by Umcaruje** [comment]
>It's a feature built into many other DAWs

Could you post an example, I've never encountered such a feature myself.

**Comment by Pyrif** [comment]
From FL Studio:
![image](https://user-images.githubusercontent.com/27585517/36055064-5f9b54b8-0dbf-11e8-90ae-90056547863b.png)

**Comment by Anonymouqs** [comment]
So basically have the ability to apply a bandwidth limit on the processes that LMMS can use up during rendering?

**Comment by PhysSong** [comment]
How about adding ability to change priority of mixer worker threads and probably `ProjectRender` thread? Using [`QThread::setPriority`](http://doc.qt.io/qt-5/qthread.html#setPriority) may be enough.

**Comment by zonkmachine** [comment]
> Using [`QThread::setPriority`](http://doc.qt.io/qt-5/qthread.html#setPriority) may be enough.

that's actually what we're doing in void [ProjectRenderer::startProcessing()](https://github.com/LMMS/lmms/blob/1a6f4c1104e8ecd469db5f27eb230994b9ee9ee2/src/core/ProjectRenderer.cpp#L157-L161). We set priority to `QThread::HighPriority` on all but win32. This sounds wrong to me. I think this should be set to a low priority, something lower than the OS default.
https://doc.qt.io/qt-5/qthread.html#Priority-enum
`QThread::LowPriority`?

**Comment by PhysSong** [comment]
> I think this should be set to a low priority, something lower than the OS default.

No, I think users should be able to choose between a low priority and a high priority. Someone may want to use higher because they want to render the project quickly, while someone like the OP will want a lower priority for running other tasks smoothly.

**Comment by zonkmachine** [comment]
If you want it to render quickly, all you need to do is not touching the computer. If you start a web browser because you get bored while rendering then do you really want the render job to have priority over the browser? Most likely not. On my slowest computer I see no difference in render speed nor is my other activity prohibited by rendering with any of the priority levels (Or the priority code didn't work..). I agree on a user setting for this but I think a good default level would be low priority for rendering for an overwhelming majority of the users.

**Comment by zonkmachine** [comment]
> Or the priority (_my_) code didn't work..)

 Changing the QThread priority probably just affected the processes within lmms and not how it's loading down the OS. I still think the idea is good though. From what I understand I think it means lowering lmms priority something like this, as snatched from main.cpp.

```
#ifdef LMMS_BUILD_WIN32
    if (!SetPriorityClass( GetCurrentProcess(), BELOW_NORMAL_PRIORITY_CLASS) )
    {
        printf( "Notice: could not set lower priority for rendering.\n" );
    }
#endif
```

**Comment by RoxasKH** [comment]
I give a +1 for this. Feels more than useful to me, and i was about to open an issue, but it was already done.

**Comment by PhysSong** [comment]
> If you want it to render quickly, all you need to do is not touching the computer.

That make sense, but what if I want to make the LMMS process prioritized than other background system applications?

**Comment by sqrvrt** [comment]
I'm half-convinced this shouldn't be a part of LMMS, but idk. Thread priorities are very OS-specific, there's whatever the hell windows uses, niceness in POSIX and scheduling policy in specifically glibc which would be the most appealing here (technically it's just not implemented in musl but those are the only main ones being used). Most of them are commonly expected to be set externally at least in other apps, or for the OS to do its job and figure stuff out.

So if a solution _is_ wanted, the way to go is probably a checkbox or similar in settings that has a bunch of OS-specific code attached to it. I'm not familiar with how exporting works so i can't exactly tell if e.g. exporting spawns a new process or not, but i laid out my point.
